Timeout for Acquire Capability in Language Server

### Discord username

_No response_

### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

To implement #7990 I need for the new GUI to acquire the capability to modify the execution context. To do so I request it via the following message (after the execution context has been created)
```
{"jsonrpc":"2.0","id":"4","method":"capability/acquire","params":{"method":"executionContext/canModify","registerOptions":{"contextId":"669445d7-96b9-4a99-a78d-739307a96175"}}}
```

This is however always answered with a 

```
{"jsonrpc":"2.0","id":"4","error":{"code":11,"message":"Request timeout"}}
```

### Expected behaviour

The request should succeed (or report an actionable error). 

### How we can reproduce it?

Run the new GUI from branch #8037 with the project manager from the latest release (2023.2.1-nightly.2023.10.8 as of time of this report)

### Screenshots or screencasts

_No response_

### Logs

```
[TRACE] [2023-10-12T14:06:36+01:00] [org.enso.jsonrpc.JsonRpcServer] Sent text message {"jsonrpc":"2.0","id":0,"result":{"projects":[{"name":"New_Project_23","namespace":"local","id":"aa1d963d-292c-409d-b1ca-d66ac0ed6ddb","created":"2023-10-12T11:37:07.236145Z","lastOpened":"2023-10-12T13:04:30.014079Z"},{"name":"New_Project_22","namespace":"local","id":"735eb773-c3b2-4cc6-8295-1e033f58cf9c","created":"2023-10-12T11:31:36.657158Z","lastOpened":"2023-10-12T11:31:37.540205Z"},{"name":"New_Project_20","namespace":"local","id":"db1cc016-b2bf-4656-aa5c-5508035b0e39","created":"2023-10-03T15:08:13.321939Z","lastOpened":"2023-10-12T11:31:00.174663Z"},{"name":"New_Project_21","namespace":"local","id":"413a67de-50db-479e-ad00-a58cc3fb1112","created":"2023-10-12T10:42:58.901288Z","lastOpened":"2023-10-12T10:42:59.9312Z"},{"name":"New_Project_19","namespace":"local","id":"43263144-496a-47f9-97d0-399f03a80bc0","created":"2023-10-03T14:50:22.587113Z","lastOpened":"2023-10-04T12:05:10.230180132Z"},{"name":"New_Project_18","namespace":"local","id":"ead3318d-7c02-4a5e-b2df-256f5a57db66","created":"2023-10-03T12:30:18.740175367Z","lastOpened":"2023-10-04T12:04:13.974964453Z"},{"name":"New_Project_17","namespace":"local","id":"450e6e1d-14d4-4841-8751-f64f37f91bb4","created":"2023-10-03T12:29:34.686917535Z","lastOpened":"2023-10-03T12:29:37.284282184Z"},{"name":" ?? ?? ?? ???? ?","namespace":"local","id":"24c75448-b965-4e70-889f-dfd904d8677c","created":"2023-07-31T11:07:33.938056Z","lastOpened":"2023-10-03T12:29:25.859869169Z"},{"name":"New_Project_16","namespace":"local","id":"088b8d6c-64cb-467c-b1ba-7febeb209f84","created":"2023-10-02T14:23:57.374541904Z","lastOpened":"2023-10-02T14:24:23.588571515Z"},{"name":"New_Project_13","namespace":"local","id":"78f5763c-2f94-4f11-9990-1ecee87ec29f","created":"2023-10-02T12:43:02.570768Z","lastOpened":"2023-10-02T14:23:56.155340218Z"},{"name":"New_Project_15","namespace":"local","id":"d949c833-4005-4bc6-8e83-e6651f5effe5","created":"2023-10-02T14:19:06.140105772Z","lastOpened":"2023-10-02T14:19:59.796047449Z"},{"name":"New_Project_14","namespace":"local","id":"6f151959-4e25-4bdc-be55-a6819d27140a","created":"2023-10-02T14:18:09.176380058Z","lastOpened":"2023-10-02T14:18:55.791660812Z"},{"name":"New_Project_12","namespace":"local","id":"116bab40-e29f-4a58-a346-f912f277021a","created":"2023-09-26T12:58:42.431492Z","lastOpened":"2023-10-02T12:42:24.756286Z"},{"name":"New_Project_11","namespace":"local","id":"a1a57df7-6fe5-44f0-be5b-6d872677e083","created":"2023-09-26T12:50:20.614555Z","lastOpened":"2023-09-26T12:57:18.756191Z"},{"name":"New_Project_10","namespace":"local","id":"7a8091b8-9f32-4381-a2c4-ffeea457c29e","created":"2023-09-26T11:52:46.278828Z","lastOpened":"2023-09-26T11:54:10.752245Z"},{"name":"New_Project_9","namespace":"local","id":"ec928f1c-e0e1-4223-a284-5173898d79db","created":"2023-09-20T15:25:41.037232Z","lastOpened":"2023-09-26T11:52:20.508647Z"},{"name":"Colorado_COVID_1","namespace":"local","id":"1796e289-cc38-4122-8b01-f7df676b6cc8","created":"2023-09-22T14:35:13.894083Z","lastOpened":null},{"name":"New_Project_8","namespace":"local","id":"159a5db4-2b5a-453c-9c1f-3fbfa9e28847","created":"2023-07-31T11:07:39.484804Z","lastOpened":"2023-08-14T11:41:13.085385Z"}]}}.
[DEBUG] [2023-10-12T14:06:36+01:00] [org.enso.projectmanager.requesthandler.ProjectListHandler] stopped
[TRACE] [2023-10-12T14:06:36+01:00] [org.enso.jsonrpc.JsonRpcServer] Sent text message {"jsonrpc":"2.0","id":1,"result":{"projects":[{"name":"New_Project_23","namespace":"local","id":"aa1d963d-292c-409d-b1ca-d66ac0ed6ddb","created":"2023-10-12T11:37:07.236145Z","lastOpened":"2023-10-12T13:04:30.014079Z"},{"name":"New_Project_22","namespace":"local","id":"735eb773-c3b2-4cc6-8295-1e033f58cf9c","created":"2023-10-12T11:31:36.657158Z","lastOpened":"2023-10-12T11:31:37.540205Z"},{"name":"New_Project_20","namespace":"local","id":"db1cc016-b2bf-4656-aa5c-5508035b0e39","created":"2023-10-03T15:08:13.321939Z","lastOpened":"2023-10-12T11:31:00.174663Z"},{"name":"New_Project_21","namespace":"local","id":"413a67de-50db-479e-ad00-a58cc3fb1112","created":"2023-10-12T10:42:58.901288Z","lastOpened":"2023-10-12T10:42:59.9312Z"},{"name":"New_Project_19","namespace":"local","id":"43263144-496a-47f9-97d0-399f03a80bc0","created":"2023-10-03T14:50:22.587113Z","lastOpened":"2023-10-04T12:05:10.230180132Z"},{"name":"New_Project_18","namespace":"local","id":"ead3318d-7c02-4a5e-b2df-256f5a57db66","created":"2023-10-03T12:30:18.740175367Z","lastOpened":"2023-10-04T12:04:13.974964453Z"},{"name":"New_Project_17","namespace":"local","id":"450e6e1d-14d4-4841-8751-f64f37f91bb4","created":"2023-10-03T12:29:34.686917535Z","lastOpened":"2023-10-03T12:29:37.284282184Z"},{"name":" ?? ?? ?? ???? ?","namespace":"local","id":"24c75448-b965-4e70-889f-dfd904d8677c","created":"2023-07-31T11:07:33.938056Z","lastOpened":"2023-10-03T12:29:25.859869169Z"},{"name":"New_Project_16","namespace":"local","id":"088b8d6c-64cb-467c-b1ba-7febeb209f84","created":"2023-10-02T14:23:57.374541904Z","lastOpened":"2023-10-02T14:24:23.588571515Z"},{"name":"New_Project_13","namespace":"local","id":"78f5763c-2f94-4f11-9990-1ecee87ec29f","created":"2023-10-02T12:43:02.570768Z","lastOpened":"2023-10-02T14:23:56.155340218Z"},{"name":"New_Project_15","namespace":"local","id":"d949c833-4005-4bc6-8e83-e6651f5effe5","created":"2023-10-02T14:19:06.140105772Z","lastOpened":"2023-10-02T14:19:59.796047449Z"},{"name":"New_Project_14","namespace":"local","id":"6f151959-4e25-4bdc-be55-a6819d27140a","created":"2023-10-02T14:18:09.176380058Z","lastOpened":"2023-10-02T14:18:55.791660812Z"},{"name":"New_Project_12","namespace":"local","id":"116bab40-e29f-4a58-a346-f912f277021a","created":"2023-09-26T12:58:42.431492Z","lastOpened":"2023-10-02T12:42:24.756286Z"},{"name":"New_Project_11","namespace":"local","id":"a1a57df7-6fe5-44f0-be5b-6d872677e083","created":"2023-09-26T12:50:20.614555Z","lastOpened":"2023-09-26T12:57:18.756191Z"},{"name":"New_Project_10","namespace":"local","id":"7a8091b8-9f32-4381-a2c4-ffeea457c29e","created":"2023-09-26T11:52:46.278828Z","lastOpened":"2023-09-26T11:54:10.752245Z"},{"name":"New_Project_9","namespace":"local","id":"ec928f1c-e0e1-4223-a284-5173898d79db","created":"2023-09-20T15:25:41.037232Z","lastOpened":"2023-09-26T11:52:20.508647Z"},{"name":"Colorado_COVID_1","namespace":"local","id":"1796e289-cc38-4122-8b01-f7df676b6cc8","created":"2023-09-22T14:35:13.894083Z","lastOpened":null},{"name":"New_Project_8","namespace":"local","id":"159a5db4-2b5a-453c-9c1f-3fbfa9e28847","created":"2023-07-31T11:07:39.484804Z","lastOpened":"2023-08-14T11:41:13.085385Z"}]}}.
[TRACE] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.JsonRpcServer] Received text message: {"jsonrpc":"2.0","id":2,"method":"project/open","params":{"projectId":"aa1d963d-292c-409d-b1ca-d66ac0ed6ddb","missingComponentAction":"Install"}}.
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.projectmanager.protocol.ClientController] now supervising Actor[akka://project-manager/user/manager-client-controller-2cda996d-44a9-47f3-af9a-2e94ce46d5c8/request-handler-ProjectOpen-b75df48a-f4f0-4bc7-80bc-89b8bbb9c8bb#-1252241629]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.projectmanager.infrastructure.log.Slf4jLogging] Opening project [aa1d963d-292c-409d-b1ca-d66ac0ed6ddb].
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.projectmanager.requesthandler.ProjectOpenHandler] started (org.enso.projectmanager.requesthandler.ProjectOpenHandler@4ef97ca)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.projectmanager.infrastructure.log.Slf4jLogging] Found project [aa1d963d-292c-409d-b1ca-d66ac0ed6ddb] in [org.enso.projectmanager.infrastructure.repository.ProjectFileRepository@744ccf1c].
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.projectmanager.infrastructure.log.Slf4jLogging] Preparing to start the Language Server for [Project(aa1d963d-292c-409d-b1ca-d66ac0ed6ddb,New_Project_23,New_Project_23,local,UserProject,2023-10-12T11:37:07.236145Z,Some(Edition(Some(2023.2.1-nightly.2023.10.8),None,Map(),Map())),/home/michael/enso/projects/New_Project_23,Some(2023-10-12T13:06:44.323365Z),Some(2023-10-12T11:37:08.11475139Z))].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] The engine [2023.2.1-nightly.2023.10.8] found.
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] Loading Graal runtime [/home/michael/.local/share/enso/runtime/graalvm-ce-java17.0.7-23.0.0].
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] Installing GraalVM components [GraalVM 23.0.0-java17.0.7, Linux].
[TRACE] [2023-10-12T14:06:44+01:00] [org.enso.runtimeversionmanager.components.GraalVMComponentUpdater] /home/michael/.local/share/enso/runtime/graalvm-ce-java17.0.7-23.0.0/bin/gu { exists=true, executable=true }
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.runtimeversionmanager.components.GraalVMComponentUpdater] Executing: JAVA_HOME=/home/michael/.local/share/enso/runtime/graalvm-ce-java17.0.7-23.0.0 GRAALVM_HOME=/home/michael/.local/share/enso/runtime/graalvm-ce-java17.0.7-23.0.0 /home/michael/.local/share/enso/runtime/graalvm-ce-java17.0.7-23.0.0/bin/gu list -v
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] Available GraalVM components: [List(GraalVMComponent(graalvm), GraalVMComponent(icu4j), GraalVMComponent(js), GraalVMComponent(llvm), GraalVMComponent(llvm-toolchain), GraalVMComponent(native-image), GraalVMComponent(python), GraalVMComponent(regex))].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] Found GraalVM runtime [GraalVM 23.0.0-java17.0.7].
[TRACE] [2023-10-12T14:06:44+01:00] [org.enso.projectmanager.infrastructure.languageserver.LanguageServerController] received handled StartServer(2cda996d-44a9-47f3-af9a-2e94ce46d5c8,Project(aa1d963d-292c-409d-b1ca-d66ac0ed6ddb,New_Project_23,New_Project_23,local,UserProject,2023-10-12T11:37:07.236145Z,Some(Edition(Some(2023.2.1-nightly.2023.10.8),None,Map(),Map())),/home/michael/enso/projects/New_Project_23,Some(2023-10-12T13:06:44.323365Z),Some(2023-10-12T11:37:08.11475139Z)),2023.2.1-nightly.2023.10.8,Actor[akka://project-manager/user/manager-client-controller-2cda996d-44a9-47f3-af9a-2e94ce46d5c8/request-handler-ProjectOpen-b75df48a-f4f0-4bc7-80bc-89b8bbb9c8bb#-1252241629],false) from Actor[akka://project-manager/temp/$o] in state supervising
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.projectmanager.requesthandler.ProjectOpenHandler] stopped
[TRACE] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.JsonRpcServer] Sent text message {"jsonrpc":"2.0","id":2,"result":{"engineVersion":"2023.2.1-nightly.2023.10.8","languageServerJsonAddress":{"host":"127.0.0.1","port":50524},"languageServerBinaryAddress":{"host":"127.0.0.1","port":52525},"projectName":"New_Project_23","projectNormalizedName":"New_Project_23","projectNamespace":"local"}}.
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] started (org.enso.jsonrpc.MessageHandlerSupervisor@67b9705d)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] now supervising Actor[akka://language-server/user/message-handler-supervisor-54fcc116-1e0b-402f-a2c3-df0b42cb7362/message-handler-b54160d2-1e4b-42b4-9614-fb0e79477b74#-1600632212]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.MessageHandler] started (org.enso.jsonrpc.MessageHandler@10bad1de)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] started (org.enso.languageserver.protocol.json.JsonConnectionController@61961f40)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now watched by Actor[akka://language-server/system/eventStreamUnsubscriber-1#1235415133]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] now watched by Actor[akka://language-server/system/Materializers/StreamSupervisor-0/$$+s#-1375848019]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.binary.BinaryConnectionController] started (org.enso.languageserver.protocol.binary.BinaryConnectionController@47fe8163)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] started (org.enso.jsonrpc.MessageHandlerSupervisor@4f5f9fc9)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] now supervising Actor[akka://language-server/user/message-handler-supervisor-4a3f5639-caac-4c10-8928-aedd0714ecbf/message-handler-2ef8f966-7e50-46af-8ece-49d985103978#1766262681]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.MessageHandler] started (org.enso.jsonrpc.MessageHandler@26a0558)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] started (org.enso.languageserver.protocol.json.JsonConnectionController@705f4142)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now watched by Actor[akka://language-server/system/eventStreamUnsubscriber-1#1235415133]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] now watched by Actor[akka://language-server/system/Materializers/StreamSupervisor-0/$$dt#-699719788]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.binary.BinaryConnectionController] now watched by Actor[akka://language-server/system/Materializers/StreamSupervisor-0/$$ct#-964351352]
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] Initializing resources.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.DirectoriesInitialization] Initializing directories...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.DirectoriesInitialization] Initialized directories.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.JsonRpcInitializationComponent] Initializing JSON-RPC protocol.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.JsonRpcInitializationComponent] JSON-RPC protocol initialized.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.ZioRuntimeInitialization] Initializing ZIO runtime...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.ZioRuntimeInitialization] ZIO runtime initialized [org.enso.languageserver.effect.ExecutionContextRuntime@53031f84].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing sql database [org.enso.searcher.sql.SqlDatabase@58f50ac4]...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.RepoInitialization] Initialized sql database [org.enso.searcher.sql.SqlDatabase@58f50ac4].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.TruffleContextInitialization] Initializing Runtime context...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.TruffleContextInitialization] Initialized Runtime context.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing suggestions repo [/home/michael/enso/projects/New_Project_23/.enso/suggestions.db]...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] Initializing resources.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.DirectoriesInitialization] Initializing directories...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.DirectoriesInitialization] Initialized directories.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.JsonRpcInitializationComponent] Initializing JSON-RPC protocol.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.binary.BinaryConnectionController] Data session initialized for client: f1eddcc1-bfcf-415c-a777-7ab19ad52595 [127.0.0.1:37384].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.RepoInitialization] Initialized Suggestions repo [/home/michael/enso/projects/New_Project_23/.enso/suggestions.db].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] RPC session initialized for client [258bfbee-d5a9-4a6f-9df8-b45263765bd0].
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now supervising Actor[akka://language-server/user/json-connection-controller-2ef8f966-7e50-46af-8ece-49d985103978/request-handler-AcquireCapability-2858a44e-47c9-463b-b7d2-47729643f9ae#-272918494]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler] started (org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler@51d66874)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler] stopped
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.JsonRpcInitializationComponent] JSON-RPC protocol initialized.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.ZioRuntimeInitialization] Initializing ZIO runtime...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.ZioRuntimeInitialization] ZIO runtime initialized [org.enso.languageserver.effect.ExecutionContextRuntime@53031f84].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing sql database [org.enso.searcher.sql.SqlDatabase@58f50ac4]...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.RepoInitialization] Initialized sql database [org.enso.searcher.sql.SqlDatabase@58f50ac4].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.TruffleContextInitialization] Initializing Runtime context...
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.TruffleContextInitialization] Initialized Runtime context.
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing suggestions repo [/home/michael/enso/projects/New_Project_23/.enso/suggestions.db]...
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now supervising Actor[akka://language-server/user/json-connection-controller-2ef8f966-7e50-46af-8ece-49d985103978/request-handler-ListFile-2fc55ba5-6218-4c90-8b6a-2e486d0b6ac2#-822720849]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.file.ListFileHandler] started (org.enso.languageserver.requesthandler.file.ListFileHandler@fa58f90)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.file.ListFileHandler] stopped
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.boot.resource.RepoInitialization] Initialized Suggestions repo [/home/michael/enso/projects/New_Project_23/.enso/suggestions.db].
[INFO] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] RPC session initialized for client [f1eddcc1-bfcf-415c-a777-7ab19ad52595].
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now supervising Actor[akka://language-server/user/json-connection-controller-2ef8f966-7e50-46af-8ece-49d985103978/request-handler-OpenFile-edaba2a3-2c1f-4808-9f53-e6acb792b158#795403765]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.text.OpenFileHandler] started (org.enso.languageserver.requesthandler.text.OpenFileHandler@33d41b99)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.text.OpenFileHandler] stopped
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now supervising Actor[akka://language-server/user/json-connection-controller-b54160d2-1e4b-42b4-9614-fb0e79477b74/request-handler-AcquireCapability-4b7f681f-7083-4f93-bb40-955a2c1bbfb0#-977427321]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler] started (org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler@4ca6d122)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler] stopped
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now supervising Actor[akka://language-server/user/json-connection-controller-b54160d2-1e4b-42b4-9614-fb0e79477b74/request-handler-ExecutionContextCreate-7b216e4f-c7c9-476a-b886-8f2b4266c2ea#1466577135]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.executioncontext.CreateHandler] started (org.enso.languageserver.requesthandler.executioncontext.CreateHandler@3fa0a5d6)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.runtime.handler.CreateContextHandler] started (org.enso.languageserver.runtime.handler.CreateContextHandler@dd087c8)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.runtime.ContextEventsListener] started (org.enso.languageserver.runtime.ContextEventsListener@63790121)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.runtime.ContextRegistry] now supervising Actor[akka://language-server/user/context-registry/$u#-1640632646]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.runtime.ContextRegistry] now supervising Actor[akka://language-server/user/context-registry/$v#-794940795]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now supervising Actor[akka://language-server/user/json-connection-controller-b54160d2-1e4b-42b4-9614-fb0e79477b74/request-handler-GetSuggestionsDatabase-e5986ce3-22d2-4e25-a3e2-daa8aceeeecc#1426458489]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.search.GetSuggestionsDatabaseHandler] started (org.enso.languageserver.requesthandler.search.GetSuggestionsDatabaseHandler@4f792904)
[DEBUG] [2023-10-12T14:06:44+01:00] [enso] Executing command asynchronously: CreateContextCmd...
[DEBUG] [2023-10-12T14:06:44+01:00] [enso] Command CreateContextCmd finished.
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.runtime.handler.CreateContextHandler] stopped
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.executioncontext.CreateHandler] stopped
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now supervising Actor[akka://language-server/user/json-connection-controller-b54160d2-1e4b-42b4-9614-fb0e79477b74/request-handler-AcquireCapability-c9b7b511-0903-48f1-ae35-3d7a3871fb20#1975559884]
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler] started (org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler@5ce5a4f3)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.search.handler.InvalidateModulesIndexHandler] started (org.enso.languageserver.search.handler.InvalidateModulesIndexHandler@5187b1dc)
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.requesthandler.search.GetSuggestionsDatabaseHandler] stopped
[DEBUG] [2023-10-12T14:06:44+01:00] [enso] Executing command asynchronously: InvalidateModulesIndexCommand...
[DEBUG] [2023-10-12T14:06:44+01:00] [enso] Command InvalidateModulesIndexCommand finished.
[DEBUG] [2023-10-12T14:06:44+01:00] [org.enso.languageserver.search.handler.InvalidateModulesIndexHandler] stopped
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.projectmanager.infrastructure.languageserver.LanguageServerSupervisor] now supervising Actor[akka://project-manager/user/language-server-registry/language-server-controller-aa1d963d-292c-409d-b1ca-d66ac0ed6ddb/supervisor-language-server-aa1d963d-292c-409d-b1ca-d66ac0ed6ddb/heartbeat-190e935d-b9b2-4218-8e02-dc5c94ca4864#-2137254536]
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.projectmanager.infrastructure.http.FanOutReceiver] started (org.enso.projectmanager.infrastructure.http.FanOutReceiver@53daae46)
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.projectmanager.infrastructure.languageserver.HeartbeatSession] Heartbeat connection initialized [Socket(127.0.0.1,50524)].
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.projectmanager.infrastructure.languageserver.HeartbeatSession] started (org.enso.projectmanager.infrastructure.languageserver.HeartbeatSession@16668b9e)
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.projectmanager.infrastructure.http.FanOutReceiver] now watched by Actor[akka://project-manager/system/Materializers/StreamSupervisor-0/$$Js#-1426391169]
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] started (org.enso.jsonrpc.MessageHandlerSupervisor@1ec41a9e)
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] now supervising Actor[akka://language-server/user/message-handler-supervisor-448632b3-f4c2-44ef-832a-5b89ce220db8/message-handler-8895df54-8765-4fc1-af98-0aea09e0ee33#1894764864]
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.jsonrpc.MessageHandler] started (org.enso.jsonrpc.MessageHandler@4ff7b686)
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] started (org.enso.languageserver.protocol.json.JsonConnectionController@7d9a3f76)
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now watched by Actor[akka://language-server/system/eventStreamUnsubscriber-1#1235415133]
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] now watched by Actor[akka://language-server/system/Materializers/StreamSupervisor-0/$$gt#476201043]
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.projectmanager.infrastructure.languageserver.HeartbeatSession] Sending ping message to Socket(127.0.0.1,50524).
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] now supervising Actor[akka://language-server/user/json-connection-controller-8895df54-8765-4fc1-af98-0aea09e0ee33/$a#-1308076503]
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.languageserver.requesthandler.monitoring.PingHandler] started (org.enso.languageserver.requesthandler.monitoring.PingHandler@6caea4e5)
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.languageserver.requesthandler.monitoring.PingHandler] stopped
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.projectmanager.infrastructure.languageserver.HeartbeatSession] Received correct pong message from Socket(127.0.0.1,50524).
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.jsonrpc.MessageHandlerSupervisor] no longer watched by Actor[akka://language-server/system/Materializers/StreamSupervisor-0/$$gt#476201043]
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.jsonrpc.MessageHandler] stopped
[DEBUG] [2023-10-12T14:06:46+01:00] [org.enso.languageserver.protocol.json.JsonConnectionController] stopped
[DEBUG] [2023-10-12T14:06:49+01:00] [org.enso.projectmanager.infrastructure.http.FanOutReceiver] no longer watched by Actor[akka://project-manager/system/Materializers/StreamSupervisor-0/$$Js#-1426391169]
[DEBUG] [2023-10-12T14:06:49+01:00] [org.enso.projectmanager.infrastructure.languageserver.HeartbeatSession] stopped
[ERROR] [2023-10-12T14:06:54+01:00] [org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler] Acquiring capability for client [f1eddcc1-bfcf-415c-a777-7ab19ad52595] timed out.
[DEBUG] [2023-10-12T14:06:54+01:00] [org.enso.languageserver.requesthandler.capability.AcquireCapabilityHandler] stopped
```

Note that there is one `AcquireCapability` request before the execution context is initialised, this is however an unrelated suggestion DB request
```
{"jsonrpc":"2.0","id":"1","method":"capability/acquire","params":{"method":"search/receivesSuggestionsDatabaseUpdates","registerOptions":{}}}
```

### Enso Version

develop

### Browser or standalone distribution

Chromium

### Browser Version or standalone distribution

Chromium Version 117.0.5938.88 (Official Build) Arch Linux (64-bit)

### Operating System

Linux

### Operating System Version

_No response_

### Hardware you are using

_No response_

--- a/engine/language-server/src/main/scala/org/enso/languageserver/capability/CapabilityRouter.scala
+++ b/engine/language-server/src/main/scala/org/enso/languageserver/capability/CapabilityRouter.scala
@@ -4,6 +4,8 @@ import akka.actor.{Actor, ActorRef, Props}
 import com.typesafe.scalalogging.LazyLogging
 import org.enso.languageserver.capability.CapabilityProtocol.{
   AcquireCapability,
+  CapabilityAcquisitionBadRequest,
+  CapabilityReleaseBadRequest,
   ReleaseCapability
 }
 import org.enso.languageserver.data.{
@@ -65,6 +67,10 @@ class CapabilityRouter(
           CapabilityRegistration(ReceivesSuggestionsDatabaseUpdates())
         ) =>
       suggestionsHandler.forward(msg)
+    case AcquireCapability(_, _) =>
+      sender() ! CapabilityAcquisitionBadRequest
+    case ReleaseCapability(_, _) =>
+      sender() ! CapabilityReleaseBadRequest
   }
 
 }

--- a/engine/language-server/src/test/scala/org/enso/languageserver/websocket/json/CapabilitiesTest.scala
+++ b/engine/language-server/src/test/scala/org/enso/languageserver/websocket/json/CapabilitiesTest.scala
@@ -1,0 +1,81 @@
+package org.enso.languageserver.websocket.json
+
+import io.circe.literal._
+import io.circe.parser.parse
+import io.circe.syntax.EncoderOps
+import org.enso.polyglot.runtime.Runtime.Api
+
+import java.util.UUID
+
+class CapabilitiesTest extends BaseServerTest {
+
+  "capability/acquire" must {
+
+    "return an error response to the client when requesting it for `executionContext/canModify`" in {
+      val client    = getInitialisedWsClient()
+      val contextId = createExecutionContext(client)
+      client.send(json"""
+        { "jsonrpc": "2.0",
+          "method": "capability/acquire",
+          "id": 1,
+          "params": {
+            "method" : "executionContext/canModify",
+            "registerOptions": {
+             "contextId" : $contextId
+            }
+          }
+        }
+        """)
+      val response = parse(client.expectMessage()).rightValue.asObject.value
+      response("jsonrpc") shouldEqual Some("2.0".asJson)
+      response("id") shouldEqual Some(1.asJson)
+      val err = response("error").value.asObject.value
+      err("message") shouldEqual Some("Service error".asJson)
+    }
+  }
+
+  "capability/release" must {
+
+    "return an error response to the client when requesting it for `executionContext/canModify`" in {
+      val client    = getInitialisedWsClient()
+      val contextId = createExecutionContext(client)
+      client.send(json"""
+        { "jsonrpc": "2.0",
+          "method": "capability/release",
+          "id": 1,
+          "params": {
+            "method" : "executionContext/canModify",
+            "registerOptions": {
+             "contextId" : $contextId
+            }
+          }
+        }
+        """)
+      val response = parse(client.expectMessage()).rightValue.asObject.value
+      response("jsonrpc") shouldEqual Some("2.0".asJson)
+      response("id") shouldEqual Some(1.asJson)
+      val err = response("error").value.asObject.value
+      err("message") shouldEqual Some("Service error".asJson)
+    }
+  }
+
+  private def createExecutionContext(client: WsTestClient): UUID = {
+    client.send(ExecutionContextJsonMessages.executionContextCreateRequest(0))
+    val (requestId, contextId) =
+      runtimeConnectorProbe.receiveN(1).head match {
+        case Api.Request(requestId, Api.CreateContextRequest(contextId)) =>
+          (requestId, contextId)
+        case msg =>
+          fail(s"Unexpected message: $msg")
+      }
+
+    runtimeConnectorProbe.lastSender ! Api.Response(
+      requestId,
+      Api.CreateContextResponse(contextId)
+    )
+    client.expectJson(
+      ExecutionContextJsonMessages.executionContextCreateResponse(0, contextId)
+    )
+    contextId
+  }
+}

